### DISCUSSION TITLE
Is it possible to connect to read data from DB2 (AS/400) database with Airflow?

### DISCUSSION THREAD

**Discussion by AdrianoNicolucci** [apache/airflow#26998] [Q&A]
**Current situation**:
I have a python script that runs on an EC2 machine that has pyodbc in python installed as well as the ibm driver in order to connect to a as/400 database. 
**Goal**:
I would like to run these scripts on a airflow server instead.
**Problem**:
 It doesn't appear that there is a "provider" extension for connecting to AS/400 databases (I could be wrong).
 **Question**:
 What do I need to do to convert my existing script in order to leverage Airflow to successfully read data from this database.

**Comment by potiuk** [comment]
If you looked through the docs and it's not there, then you can use PythonOperator (and similar to run any code of yours. That's pretty much it. If you want you can encapsulate it in custom operator (just search for custom operator in Airflow docs). Writing custom operator is a first-class citizen in Airlfow. Also you can look if someone has not written their custom provider for AS/400  - maybe someone did but it needs looking for it - there are places like astronomer's registry and plain google search that can be used for it.
